### ISSUE TITLE
Support a special view in the quick pick optimized for displaying icons

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#192309]
The terminal has this existing icon selector:

![Image](https://github.com/microsoft/vscode/assets/2193314/8df97470-a97d-4e3b-b3d8-d470604ba725)

There are several problems with this though:

- Internal codicon details are exposed, for example there are 4 plus icons at the start
- Icons are not translated
- All codicons are included, meaning themed icons like red debug breakpoints are included
- Picking an icon visually is a headache as you need to either search for what you think the icon is called or look through the whole list

@hbons put together this prototype exploring an icon selector for profiles:

![Image](https://github.com/microsoft/vscode/assets/2193314/738de5a9-baab-46b4-9b0a-f4251045b595)

I propose we implement @hbons's mock up above, but do it in such a way that the quick pick can be opened in a special mode that shows icons only initially. Then when filtering, it could revert to the standard display closer to the terminal's current icon picker so that the filtered search text is visible and it acts like a regular quick pick.

Doing it this way as opposed to creating a whole new UI component allows us to share code, enhance the quick pick, potentially expose this new mode as an extension API and keep the UX as consistent as possible.

**Comment by TylerLeonhardt** [comment]
Yeah this makes sense to me. We use a list for QuickPick today... so we'd still need some sort of a grid thing? If we don't have that already?

https://github.com/microsoft/vscode/blob/e682a8dcd2de106b6250051698c3bf50b21de6bb/src/vs/platform/quickinput/common/quickInput.ts#L278-L388

Here's what `QuickPick` looks like today... maybe we have an option called `iconOnly` or similar to only render the icons of `QuickPickItem`s.

**Comment by sandy081** [comment]
@TylerLeonhardt I would like to work on this because it is needed for profiles icon feature. I will come up with a PR. Hope this is fine.

**Comment by Tyriar** [comment]
> maybe we have an option called iconOnly or similar to only render the icons of QuickPickItems.

I think filtering should revert to the regular view so users can see the text/ids behind the scenes, this makes also helps tie the experiences together in the mind of the user. If that's the approach we take, something more along the lines of an icon preferred/focused view would be better than using "only" as the option.

**Comment by TylerLeonhardt** [comment]
> I think filtering should revert to the regular view so users can see the text/ids behind the scenes,

Idk... Slack doesn't do that:

![image](https://github.com/microsoft/vscode/assets/2644648/20c1403d-10ee-4816-8354-a7386a418162)

nor does Teams:
![image](https://github.com/microsoft/vscode/assets/2644648/34609144-6117-438a-a647-236509e6e3f3)

However, they both support showing the id on hover which we could do.

**Comment by Tyriar** [comment]
@TylerLeonhardt it would be better if the emoji picker did do that imo, otherwise it's difficult knowing what text is being searched especially without using the mouse (and vscode users like to work with text and the keyboard).

**Comment by Tyriar** [comment]
@TylerLeonhardt if it was just the regular quick pick it suffers from the issues where it's difficult to find an icon if you don't know what you're looking for as described in the top comment

**Comment by sandy081** [comment]
Throwing another idea: How about supporting in place context widget like below?

<img width="606" alt="image" src="https://github.com/microsoft/vscode/assets/10746682/a97a33d0-eec8-4e27-a329-2b59e1ab0c7c">

If we have a profile management editor, then this will give better user experience.

**Comment by hbons** [comment]
@sandy081 We could do that when we have local QuickPick mode. See #189030. For now I'd hold off creating a new control.

**Comment by hbons** [comment]
Agree with @TylerLeonhardt it would be nicer to keep the grid mode when searching. We could show the non-matching icons in disabled state instead of hiding them.

**Comment by TylerLeonhardt** [comment]
Coming back to do this post-UX sync... I agree with @Tyriar that resorting to the list view when the user types something in the input is ideal. Mostly because:
* right arrow & left arrow keys will have a more aligned behavior
* the user can see what the label of the icon is

**Comment by TylerLeonhardt** [comment]
Moving this to backlog since this is a feature request of the quick pick and I'm not planning on picking this up at the moment.

**Comment by mrleemurray** [comment]
Nope, was just checking.
